vault/wrapping.go

```diff
@@ -309,16 +309,57 @@ DONELISTHANDLING:
 	return nil, nil
 }
 
-// ValidateWrappingToken checks whether a token is a wrapping token.
-func (c *Core) ValidateWrappingToken(ctx context.Context, req *logical.Request) (bool, error) {
+// validateWrappingToken checks whether a token is a wrapping token. The passed
+// in logical request will be updated if the wrapping token was provided within
+// a JWT token.
+func (c *Core) ValidateWrappingToken(ctx context.Context, req *logical.Request) (valid bool, err error) {
+	defer func() {
+		// Perform audit logging before returning if there's an issue with checking
+		// the wrapping token
+		if err != nil || !valid {
+			// Get non-HMAC'ed request data keys
+			var nonHMACReqDataKeys []string
+			entry := c.router.MatchingMountEntry(ctx, req.Path)
+			if entry != nil {
+				// Get and set ignored HMAC'd value.
+				if rawVals, ok := entry.synthesizedConfigCache.Load("audit_non_hmac_request_keys"); ok {
+					nonHMACReqDataKeys = rawVals.([]string)
+				}
+			}
+
+			// We log the Auth object like so here since the wrapping token can
+			// come from the header, which gets set as the ClientToken
+			auth := &logical.Auth{
+				ClientToken: req.ClientToken,
+				Accessor:    req.ClientTokenAccessor,
+			}
+
+			logInput := &logical.LogInput{
+				Auth:               auth,
+				Request:            req,
+				NonHMACReqDataKeys: nonHMACReqDataKeys,
+			}
+			if err != nil {
+				logInput.OuterErr = errwrap.Wrapf("error validating wrapping token: {{err}}", err)
```

Comment:
Are we confident that `{{err}}` will always be safe to include un-HMACed in the audit log?  I think it's probably safer to omit the cause.